src/OrchardCore/OrchardCore.Localization.Core/PortableObject/PoParser.cs

```diff
@@ -3,224 +3,235 @@
-    /// Represents a parser for portable objects.
+    /// Parses a .po file.
     /// </summary>
-    public class PoParser
-    {
-        private static readonly Dictionary<char, char> _escapeTranslations = new()
-        {
-            { 'n', '\n' },
-            { 'r', '\r' },
-            { 't', '\t' },
-        };
+    /// <param name="reader">The <see cref="TextReader"/>.</param>
+    /// <returns>A list of culture records.</returns>
+    [Obsolete("This methos has been deprecated, please use ParseAsync instead.")]
+    public IEnumerable<CultureDictionaryRecord> Parse(TextReader reader) => ParseAsync(reader).GetAwaiter().GetResult();
 
-        /// <summary>
-        /// Parses a .po file.
-        /// </summary>
-        /// <param name="reader">The <see cref="TextReader"/>.</param>
-        /// <returns>A list of culture records.</returns>
+    /// <summary>
+    /// Parses a .po file.
+    /// </summary>
+    /// <param name="reader">The <see cref="TextReader"/>.</param>
+    /// <returns>A list of culture records.</returns>
 #pragma warning disable CA1822 // Mark members as static
-        public IEnumerable<CultureDictionaryRecord> Parse(TextReader reader)
+    public async Task<IEnumerable<CultureDictionaryRecord>> ParseAsync(TextReader reader)
 #pragma warning restore CA1822 // Mark members as static
+    {
+        var entryBuilder = new DictionaryRecordBuilder();
+        var cultureDictionaryRecords = new List<CultureDictionaryRecord>();
+        string line;
+        while ((line = await reader.ReadLineAsync()) != null)
         {
-            var entryBuilder = new DictionaryRecordBuilder();
-            string line;
-            while ((line = reader.ReadLine()) != null)
-            {
-                (var context, var content) = ParseLine(line);
-
-                if (context == PoContext.Other)
-                {
-                    continue;
-                }
-
-                // msgid or msgctxt are first lines of the entry. If builder contains valid entry return it and start building a new one.
-                if ((context == PoContext.MessageId || context == PoContext.MessageContext) && entryBuilder.ShouldFlushRecord)
-                {
-                    yield return entryBuilder.BuildRecordAndReset();
-                }
+            (var context, var content) = ParseLine(line);
 
-                entryBuilder.Set(context, content);
+            if (context == PoContext.Other)
+            {
+                continue;
             }
 
-            if (entryBuilder.ShouldFlushRecord)
+            // msgid or msgctxt are first lines of the entry. If builder contains valid entry return it and start building a new one.
+            if ((context == PoContext.MessageId || context == PoContext.MessageContext) && entryBuilder.ShouldFlushRecord)
             {
-                yield return entryBuilder.BuildRecordAndReset();
+                cultureDictionaryRecords.Add(entryBuilder.BuildRecordAndReset());
```

Comment:
Keep the yield and return `IAsyncEnumerable`.

Reply:
It will not be awaitable, right?

Reply:
It will be awaitable, that's the point of `IAsyncEnumerable`.